multiqc/modules/biscuit/biscuit.py

```diff
@@ -64,214 +66,210 @@ def __init__(self):
+            ("covdist_all_base_botgc", parse_covdist),
+            # Coverage distribution - all bases
+            ("covdist_all_base", parse_covdist),
+            # Coverage distribution - all bases, top GC-content
+            ("covdist_all_base_topgc", parse_covdist),
+            # Coverage distribution - q40 bases, bottom GC-content
+            ("covdist_q40_base_botgc", parse_covdist),
+            # Coverage distribution - q40 bases
+            ("covdist_q40_base", parse_covdist),
+            # Coverage distribution - q40 bases, top GC-content
+            ("covdist_q40_base_topgc", parse_covdist),
+            # Coverage distribution - all cpgs, bottom GC-content
+            ("covdist_all_cpg_botgc", parse_covdist),
+            # Coverage distribution - all cpgs
+            ("covdist_all_cpg", parse_covdist),
+            # Coverage distribution - all cpgs, top GC-content
+            ("covdist_all_cpg_topgc", parse_covdist),
+            # Coverage distribution - q40 cpgs, bottom GC-content
+            ("covdist_q40_cpg_botgc", parse_covdist),
+            # Coverage distribution - q40 cpgs
+            ("covdist_q40_cpg", parse_covdist),
+            # Coverage distribution - q40 cpgs, top GC-content
+            ("covdist_q40_cpg_topgc", parse_covdist),
+            # CpG Retention by Position in Read
+            ("cpg_retention_readpos", parse_retention_readpos),
+            # CpH Retention by Position in Read
+            ("cph_retention_readpos", parse_retention_readpos),
+            # Base-averaged cytosine retention rate
+            ("base_avg_retention_rate", parse_avg_retention),
+            # Read-averaged cytosine retention rate
+            ("read_avg_retention_rate", parse_avg_retention),
         ]
 
-        # Find and parse alignment reports
-        for k in self.mdata:
-            for f in self.find_log_files(f"biscuit/{k}"):
-                s_name = f["fn"]
-                for suffix in file_suffixes:
-                    s_name = s_name.replace(suffix, "")
-                s_name = self.clean_s_name(s_name, f)
-
-                # Add source file to multiqc_sources.txt
-                self.add_data_source(f, s_name=s_name, section=k)
-
-                if s_name in self.mdata[k]:
-                    log.debug(f"Duplicate sample name found in {f['fn']}! Overwriting: {s_name}")
-
-                self.mdata[k][s_name] = getattr(self, f"parse_logs_{k}")(f["f"], f["fn"])
-
-        for k in self.mdata:
-            self.mdata[k] = self.ignore_samples(self.mdata[k])
-
-        n_samples = max([len(self.mdata[k]) for k in self.mdata])
+        n_covdist_samples = 0
+        for file_type, parser_func in file_types:
+            collected_data = {}
+            for f in self.find_log_files(f"biscuit/{file_type}"):
+                parsed_data = parser_func(f["f"], f["fn"])
+                if parsed_data is not None:
+                    s_name = self.clean_s_name(f["s_name"], f)
```

Comment:
Given how the parsing works, I think it might make sense to check ignored sample names here and bail early:

```suggestion
                    s_name = self.clean_s_name(f["s_name"], f)
                    if self.is_ignore_sample(s_name):
                        continue
```

This way, ignored samples don't get added to data sources, and we also avoid weird edge cases to do with overwriting with ignored samples etc.